bombastic/indexer/src/lib.rs

```diff
@@ -53,11 +55,16 @@ impl Run {
                 "bombastic-indexer",
                 |_context| async { Ok(()) },
                 |context| async move {
-                    let sbom_index: Box<dyn WriteIndex<Document = (SBOM, String)>> =
-                        Box::new(bombastic_index::Index::new());
+                    let sbom_index: Box<dyn WriteIndex<Document = (SBOM, String)>> = Box::new(sbom::Index::new());
                     let index = block_in_place(|| {
                         IndexStore::new(&self.storage, &self.index, sbom_index, context.metrics.registry())
                     })?;
+
+                    let package_index: Box<dyn WriteIndex<Document = (SBOM, String)>> =
+                        Box::new(packages::Index::new());
+                    let package_store = block_in_place(|| {
+                        IndexStore::new(&self.storage, &self.index, package_index, &Registry::new())
```

Comment:
We need to fix this, as those metrics won't be exposed.

Reply:
like this one https://github.com/tikv/rust-prometheus/blob/master/examples/example_custom_registry.rs#L24?

Reply:
It is better to prefix the metrics in register() taking the name as a parameter. So that in Metrics::register() the registration becomes:


```
  fn register(registry: &Registry, prefix: &str) -> Metrics {
   let indexed_total = register_int_counter_with_registry!(
            opts!("{prefix}_index_indexed_total", "Total number of indexing operations"),
            registry
        )?;


```

Reply:
after create register with prefix, the prefix will be in here.  https://github.com/tikv/rust-prometheus/blob/master/examples/example_custom_registry.rs#L24

Reply:
I believe that creating a new registry would also mean that an additional one needs to be exposed. So re-using the same instance makes more sense to me.

Reply:
Sorry, Ulf already wrote that "below".